File: app/src/main/java/co/edu/unipiloto/myapplication/ui/WelcomeActivity.java
package co.edu.unipiloto.myapplication.ui;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import co.edu.unipiloto.myapplication.R;
// Importamos MaterialButton, ya que lo usamos en el XML
import com.google.android.material.button.MaterialButton;

/**
 * Actividad de bienvenida que presenta las opciones de inicio de sesión para diferentes roles de usuario,
 * además de la opción de rastreo de envíos para el público general.
 */
public class WelcomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        // Opciones principales (usando MaterialButton para coincidir con el XML dinámico)
        MaterialButton btnCheckStatus = findViewById(R.id.btnCheckStatus); // Botón NARANJA: Consultar Estado
        MaterialButton btnRequestShipping = findViewById(R.id.btnRequestShipping); // Botón AZUL: Solicitar Envío

        // Opciones de Roles (usando MaterialButton para consistencia)
        MaterialButton btnDrivers = findViewById(R.id.btnDrivers);
        MaterialButton btnOfficials = findViewById(R.id.btnOfficials);
        Button btnAdmin = findViewById(R.id.btnAdmin); // El botón de texto discreto puede ser Button o MaterialButton.

        // 1. 🚀 ACCIÓN DINÁMICA: CONSULTAR ESTADO (NARANJA SÓLIDO) 🚀
        // Abre la nueva pantalla de rastreo para el público.
        btnCheckStatus.setOnClickListener(v -> {
            Intent i = new Intent(WelcomeActivity.this, TrackShippingActivity.class);
            startActivity(i);
        });

        // 2. Botón para usuarios/clientes que solicitan envío (AZUL SÓLIDO)
        btnRequestShipping.setOnClickListener(v -> {
            Intent i = new Intent(WelcomeActivity.this, LoginActivity.class);
            startActivity(i);
        });

        // 3. Botón para conductores (VERDE DELINEADO)
        btnDrivers.setOnClickListener(v -> {
            Intent i = new Intent(WelcomeActivity.this, LoginDriverActivity.class);
            startActivity(i);
        });

        // 4. Botón para funcionarios (VERDE DELINEADO)
        btnOfficials.setOnClickListener(v -> {
            // Nota: Aquí podrías necesitar un Login diferente (ej. LoginOfficialActivity) si las credenciales son distintas a las del cliente.
            Intent i = new Intent(WelcomeActivity.this, LoginActivity.class);
            startActivity(i);
        });

        // 5. Botón para gestión de administradores (Texto discreto)
        btnAdmin.setOnClickListener(v -> {
            Intent i = new Intent(WelcomeActivity.this, AdminPanelActivity.class);
            startActivity(i);
        });
    }
}

File: app/src/main/java/co/edu/unipiloto/myapplication/ui/WelcomeActivity.kt
package co.edu.unipiloto.myapplication.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import co.edu.unipiloto.myapplication.R
import co.edu.unipiloto.myapplication.storage.SessionManager
import com.google.android.material.button.MaterialButton
import co.edu.unipiloto.myapplication.ui.*


/**
 * Activity de Bienvenida, mapeada al layout activity_main.xml.
 * Es la primera pantalla que aparece, maneja la redirección de sesión activa
 * y la navegación a Login, Registro, Rastreo, y accesos específicos.
 */
class WelcomeActivity : AppCompatActivity() {

    private lateinit var sessionManager: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // Usamos activity_main.xml para el welcome/landing

        sessionManager = SessionManager(this)

        // 1. Verificar sesión activa y redirigir
        if (sessionManager.isLoggedIn()) {
            navigateToDashboard(sessionManager.getRole())
            return
        }

        // 2. Configurar Listeners
        setupListeners()
    }

    private fun setupListeners() {
        // Mapeo de botones de activity_main.xml
        findViewById<MaterialButton>(R.id.btnCheckStatus).setOnClickListener {
            // Ir a la pantalla de rastreo (TrackShippingActivity)
            startActivity(Intent(this, TrackShippingActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.btnRequestShipping).setOnClickListener {
            // Ir a la pantalla de Login (para Clientes)
            startActivity(Intent(this, LoginActivity::class.java))
        }

        // Acceso de Personal
        findViewById<MaterialButton>(R.id.btnOfficials).setOnClickListener {
            // Funcionarios/Personal de Sucursal -> Usan el Login ÚNICO
            startActivity(Intent(this, LoginActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.btnDrivers).setOnClickListener {
            // Conductores -> Usan el Login ÚNICO
            startActivity(Intent(this, LoginActivity::class.java))
        }

        findViewById<MaterialButton>(R.id.btnAdmin).setOnClickListener {
            // Administración/Analista -> Usan el Login ÚNICO
            startActivity(Intent(this, LoginActivity::class.java))
        }
    }

    private fun navigateToDashboard(role: String) {
        val destClass = when (role.uppercase()) {
            "CLIENTE" -> MainActivity::class.java
            "CONDUCTOR" -> DriverDashboardActivity::class.java
            "GESTOR" -> ManagerDashboardActivity::class.java
            "FUNCIONARIO" -> BranchDashboardActivity::class.java
            "ANALISTA" -> AdminPanelActivity::class.java
            else -> LoginActivity::class.java
        }
        val intent = Intent(this, destClass)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        finish()
    }
}